features/step_definitions/tag_steps.rb

```diff
@@ -102,6 +102,10 @@
   rel.add_association(fand)
 end
 
+Given "the non(-)canonical media {string}" do |tag_name|
+  Media.find_or_create_by_name(tag_name)
+end
+
 Given /^a (non-?canonical|canonical) (\w+) "([^\"]*)"$/ do |canonical_status, tag_type, tag_name|
```

Comment:
With inflection rules, you can reuse this step instead of creating a separate one for media.